packages/store/src/index.ts

```diff
@@ -0,0 +1,231 @@
+ * @returns {
+ *  {T} state - The current state of the store.
+ *  {G} getters - Predefined methods to access the state of the store.
+ *  {A} actions - Predefined methods to mutate the state of the store.
+ *  {SetStoreFunction<T>} setState - A function to manually mutate the current state of the store.
+ * }
+ */
+export type SharedStore<T extends object, G = undefined, A = undefined> = {
+  state: T;
+  getters: undefined extends G ? undefined : G;
+  actions: undefined extends A ? undefined : A;
+  setState: SetStoreFunction<T>;
+};
+
+/**
+ * Factory to create reusable stores.
+ *
+ * @typedef StoreFactory
+ * @template <T: state type of the store, G: store access methods, A: store mutation methods>
+ * @returns (initialValueSetter?: T | ((fallbackState: T) => T)) => SharedStore<T, G, A>
+ */
+export type StoreFactory<T extends object, G = undefined, A = undefined> = (
+  initialValueSetter?: T | ((fallbackState: T) => T)
+) => SharedStore<T, G, A>;
+
+/**
+ * createSharedStore takes an `initialState` and an optional object containing functions to create getters and/or actions.
+ * A store is created and returned along with access and mutation functions & the native `setState`.
+ * @template <T: state type of the store, G: store access methods, A: store mutation methods>
+ * @param initialState object to be wrapped in a Store
+ * @param createMethods optional object containing functions to create getters and/or actions.
+ * @returns `SharedStore`
+ * @example
+ * ```tsx
+ * // `counter-store.ts`
+ * export const counterStore = createSharedStore({
+ *   value: 5,
+ * }, {
+ *  getters: state => ({
+ *   count: () => state.value,
+ *  }),
+ *  actions: (setState, state) => ({
+ *   increment: () => setState(val => ({ ...val, value: val.value + 1 })),
+ *   reset: () => setState("value", 0),
+ *  })
+ * });
+ *
+ * // `Example.tsx`
+ * import { counterStore } from './counter-store.ts';
+ *
+ * const {state: counterState, getters: {count}, actions: {increment, reset}} = counterStore;
+ * count() // => 5
+ * increment()
+ * count() // => 6
+ * reset() // => 0
+ * ```
+ */
+export function createSharedStore<
+  T extends object,
+  G extends object | undefined,
+  A extends object | undefined
+>(
+  initialState: T,
+  createMethods: {
+    getters: (state: T) => G;
+    actions: (setState: SetStoreFunction<T>, state: T) => A;
+  }
+): SharedStore<T, G, A>;
+export function createSharedStore<T extends object, G extends object | undefined>(
+  initialState: T,
+  createMethods: {
+    getters: (state: T) => G;
+  }
+): SharedStore<T, G>;
+export function createSharedStore<T extends object, A extends object | undefined>(
+  initialState: T,
+  createMethods: {
+    actions: (setState: SetStoreFunction<T>, state: T) => A;
+  }
+): SharedStore<T, undefined, A>;
+export function createSharedStore<T extends object>(
+  initialState: T,
+  createMethods?: {}
+): SharedStore<T>;
+export function createSharedStore<
```

Comment:
I know that I kept the `createSharedStore` in my examples, but it still doesn't convey the function.
Maybe `createFluxStore`? At least it hints at the use of actions.
I don't really have much better ideas, unfortunately.

Reply:
What about a naming convention building on the existing function name? Since we're using `createStore` to create a store library, what do you think about `createStoreLibrary`? 

Here are those two names with a few last minute ideas:
-   ***createFluxStore***
-   ***createStoreLibrary***
-   createState
-   createStateStore
-   createStateLibrary
-   createStoreDefinition
-   createComposableStore

What do you think?

Reply:
We take a normal solid store, organize reads to state and getters, and write to actions.
My other idea would be `createImplStore` as the user includes the implementation of state management during creation.